File: c199.py
def main():
    while True:
        try:
            N = input().split(' ')
        except:
            break

        N = N[1:len(N)]
        
        i = 1
        hilltop = 0
        while True:
            if i < len(N)-1:
                if int(N[i-1]) < int(N[i]) and int(N[i]) > int(N[i+1]):
                    hilltop += 1
                elif int(N[i-1]) < int(N[i]) == int(N[i+1]):
                    temp = int(N[i])
                    while temp == int(N[i]):
                        if i < len(N)-1:
                            i += 1
                        else:
                            break
                    if int(N[i]) < temp:
                        hilltop += 1
                i += 1
            else:
                break

        print(hilltop)


main()

File: IOI&APCS/c199/c199.cpp
#include <iostream>

using namespace std;

int main(){
    int n, i, j, k = 1, hilltop = 0, temp, previous;
    int L[10001];
    while(scanf("%d", &n) != EOF){
        j = 0, previous = -1;
        for(i = 0; i < n; i++){
            scanf("%d", &temp);
            if(previous == temp)
                continue;
            else{
                L[j] = temp;
                j++;
            }
            previous = temp;
        }
        while(1){
            if(k < j-1){
                if(L[k-1] < L[k] && L[k] > L[k+1])
                    hilltop++;
                k++;
            }
            else
                break;
        }

        cout << hilltop << endl;
        k = 1;
        hilltop = 0;


    }
    return 0;
}
